Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -804,6 +804,7 @@ void RenderElement::styleWillChange(StyleDifference diff, const RenderStyle& new
         if (contentVisibilityChanged) {
             if (oldStyle->contentVisibility() == ContentVisibility::Auto)
                 ContentVisibilityDocumentState::unobserve(*element());
+            ContentVisibilityDocumentState::elementContentVisibilityChange(*element(), oldStyle->contentVisibility(), newStyle.contentVisibility());
```

Comment:
I'm not super familiar with the animations code in general, but I find it strange that we're doing this based on content visibility style rather than the effective skipped content style. Seems like the spec only mentions skipped content for animations?

Is this something we have test coverage for?

Reply:
@rwlbuis If the c-v root has c-v: auto and changes the content relevancy from relevant to irrelevant, and vice-versa, is that covered? The c-v property isn't changing in that specific case

Reply:
You are right, I missed that situation because there was no test. I now added the test and needed. code change.